### PR TITLE
Add config option to show/suppress deposit agreement

### LINKED ISSUE #435: Add ability to disable deposit agreement functionality entirely.
Our tool is internal, and much of the content that gets ingested is public domain, making this a confusing thing to have on the form.

### PR DESCRIPTION
**Pull request by escowles**
Fixes #435 

Adds a configuration option to completely disable the deposit agreement.

@projecthydra-labs/hyrax-code-reviewers

### CODE REVIEW

File: lib/hyrax/configuration.rb [hunk 1]
```diff
@@ -319,6 +319,11 @@ def active_deposit_agreement_acceptance
       @active_deposit_agreement_acceptance
     end
 
+    attr_writer :show_deposit_agreement
+    def show_deposit_agreement
+      @show_deposit_agreement || true
```
Comment (mjgiarlo) [inline lib/hyrax/configuration.rb]: If `@show_deposit_agreement` is `false`, won't this method return `true`?

Comment (escowles) [inline lib/hyrax/configuration.rb]: Dang, you're right — update coming shortly.

File: lib/hyrax/configuration.rb [hunk 2]
```diff
@@ -319,6 +319,12 @@ def active_deposit_agreement_acceptance
       @active_deposit_agreement_acceptance
     end
 
+    attr_writer :show_deposit_agreement
+    def show_deposit_agreement
```
Comment (jcoyne) [inline lib/hyrax/configuration.rb]: Please change this to `show_deposit_agreement?`

Comment (escowles) [inline lib/hyrax/configuration.rb]: There are a bunch of existing boolean options `enable_ffmpeg`, `display_microdata`, etc., and none of them have question marks now.  Should all of them have question marks?

Comment (jcoyne) [inline lib/hyrax/configuration.rb]: Yes, imho. But you don't have to handle that if you don't want to.

**Comment by jcoyne** [comment]
Would this be better for flipflop?

File: lib/hyrax/configuration.rb (hunk 2 above)
Comment (escowles) [inline lib/hyrax/configuration.rb]: OK, I've updated to `show_deposit_agreement?` and created #508 for changing the rest of the boolean getter methods to also end with question marks.

**Comment by jcoyne** [comment]
@escowles don't you have the ability to set the flipflop defaults via a yaml file?  I can see that this would be something that someone who uses a Hyku product (no access to change configuration code) would want.

**Comment by escowles** [comment]
Closing in favor of #510

**Pull request closed without merging**